src/Players/util.cpp

```diff
@@ -1,15 +1,51 @@
 #include "hex-ai/Players/util.hpp"
 #include <iostream>
+#include <limits>
 
 using Action = GameState::HexState::Action;
 
 void Players::get_action(Action &a) {
     int t;
-    std::cout << "Enter x coordinate: ";
-    std::cin >> t;
-    a.x = t;
-    std::cout << "Enter y coordinate: ";
-    std::cin >> t;
-    a.y = t;
+
+    // every darn time the user enters the wrong number, we correct them
+    for (;;) { 
+
+        std::cout << "Enter x coordinate: ";
+
+        // store user input in temp variable
+        std::cin >> t;
+
+        if (std::cin.fail()) {
+
+            // notify user
+            std::cout << "Input should be an integer between 0 and 10 \n";
+
+            // now remove `failbit` (error flag) to reset stream state
+            std::cin.clear(); 
+
+            // ignore entered characters (up to first of max len of streamsize || '\n')
+            std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n'); 
+        } else if (t < 0 || t > 10) {
```

Comment:
Oh also I feel like it would be good to have this use the macro for the board size (and every other instance of that in this code as well)